types/part_set.go

```diff
@@ -270,9 +270,12 @@ func (ps *PartSet) Total() uint32 {
 }
 
 func (ps *PartSet) AddPart(part *Part) (bool, error) {
+	// TODO: remove this? would be preferable if this only returned (false, nil)
+	// when its a duplicate block part
 	if ps == nil {
 		return false, nil
```

Comment:
I checked the code and this function is only called from `State`'s `addProposalBlockPart` function (ignoring test code here).
If you look at `addProposalBlockPart`'s code immediately following the call to `AddPart`, you see that it is acting as though the part had indeed been added. So, returning `false, nil` in this line is a BUG. We must return `false` and an error, such as `fmt.Errorf("nil PartSet")`

Reply:
not sure I understand this, my argument was that this check is redundant since before we call AddPart we already check if the part set is nil!